### PR TITLE
fix: reload log level on SIGHUP

### LINKED ISSUE #943: Reload log level on `SIGHUP`
## What to build

On `SIGHUP`, re-read the configured log level and apply it to the running process live instead of treating `SIGHUP` as a shutdown signal.

Today `SIGHUP` is registered alongside `SIGINT`/`SIGTERM` in `cmd/root_cmd.go` and cancels the root context (shutdown).

## Requirements:
  - [x] `SIGHUP` re-reads logging.level from the config file and applies it to the running logger with no restart and no shutdown
  - [x] An invalid level is rejected and the previous level keeps running (never crash the process)
  - [x] The level change is logged at a level that is always visible
  - [x] Existing `SIGINT`/`SIGTERM` shutdown behavior is unchanged
  - [x] `SIGHUP` with no change in level is a no-op

### PR DESCRIPTION
**Pull request by pollychen-lab**
## Summary

Reload the active log level when pgstream receives `SIGHUP` instead of treating it as a shutdown signal.

## Related Issue

Fixes #943

## Changes

- Handles `SIGHUP` separately from shutdown signals.
- Reloads config and applies a changed logger level through zerolog's global level filter.
- Rejects invalid log levels and keeps the previous logger level in place.
- Avoids writing config reload notices to stdout during `SIGHUP` reloads.
- Adds focused tests for successful reload, invalid-level preservation, and quiet reload behavior.

## Testing

- [x] `docker run --rm -v "$PWD":/src -w /src golang:1.26.4 gofmt -w cmd/root_cmd.go cmd/run_cmd.go cmd/snapshot_cmd.go cmd/root_cmd_test.go cmd/config/config.go internal/log/zerolog/zerolog.go pkg/log/zerolog/logger.go`
- [x] `docker run --rm -v "$PWD":/src -w /src golang:1.26.4 go test ./cmd/... ./internal/log/... ./pkg/log/...`
- [x] `docker run --rm -v "$PWD":/src -w /src golang:1.26.4 go test -race ./cmd`
- [x] `git diff --check`

## Notes

Local host does not have Go installed, so validation was run in Docker with `golang:1.26.4`, matching the repository's Go version.

### CODE REVIEW

File: cmd/root_cmd.go [hunk 1]
```diff
@@ -147,23 +150,81 @@ func Execute() error {
...
+	}()
+}
+
+func logLevelReloads(ctx context.Context) <-chan os.Signal {
+	reloads, _ := ctx.Value(sighupReloadsContextKey{}).(<-chan os.Signal)
+	return reloads
+}
+
+func reloadLogLevel(logger *rszerolog.Logger) error {
+	if err := config.Load(); err != nil {
```
[the remark this answers is not in the document]
Comment (pollychen-lab) [reply cmd/root_cmd.go]: Good catch. I added a quiet config reload path and switched the SIGHUP handler to use that, so reloading a config file no longer emits the `using config file...` notice to stdout. I also added a regression test that captures stdout during reload and verifies it stays empty.

File: cmd/root_cmd.go [hunk 2]
```diff
@@ -147,23 +150,81 @@ func Execute() error {
...
+
+	level, err := rszerolog.ParseLevel(cfg.LogLevel)
+	if err != nil {
+		return err
+	}
+	if logger.GetLevel() == level {
+		return nil
+	}
+
+	*logger = logger.Level(level)
```
[the remark this answers is not in the document]
Comment (pollychen-lab) [reply cmd/root_cmd.go]: That makes sense. I changed the reload path so it no longer overwrites the shared `*zerolog.Logger`. The logger now keeps a stable trace-level instance and reload updates zerolog’s global level filter instead, which avoids mutating the logger struct while other goroutines may be logging. I also ran `go test -race ./cmd` successfully.

**Review by kvch — APPROVED** [review]

**Comment by kvch** [comment]
Thank you!
